src/LiveComponent/assets/dist/live_controller.js

```diff
@@ -2152,6 +2152,7 @@ class Component {
         if (!controls.shouldRender) {
             return;
         }
+        this.hooks.triggerHook('loading.state:finished', this.element);
         if (backendResponse.response.headers.get('Location')) {
             if (this.isTurboEnabled()) {
                 Turbo.visit(backendResponse.response.headers.get('Location'));
```

Comment:
I'm not sure we want to trigger the "finished" event _before_ Turbo does its work.

Here you could do something just before the 

```js
window.location.href = backendResponse.response.headers.get('Location') || '';
```

Reply:
I use Turbo on this project and I need it to be triggered in that setup too. BTW, I'm actualy using this patch and it didn't seems to create any issues.